core/services/vagrant_service.rb

```diff
@@ -7,16 +7,42 @@
 module VagrantService
   include ReturnCodes
 
-  def self.chek_vmlinuz_access_rights
-    Dir.glob("/boot/vmlinuz*").each do |file|
-      stat = File.stat(file)
-      if (stat.mode & 0o004) == 0
-        return false
+  def self.set_access_rights_for_ubuntu_or_mint(logger)
+    if self.chek_distro_is_ubuntu_or_mint
+      Dir.glob("/boot/vmlinuz*").each do |file|
+        stat = File.stat(file)
+        if (stat.mode & 0o004) == 0
```

Comment:
The if statements are too deep. Please refactor the method with the following in mind.

Firstly use if-guard statements. I.e. return from the function if the distributive is not in Debian family.

Secondly extract loop check for file permissions into a separate function that indicates whether the change is required.

Lastly, clean-up the remaining logic. If possible, use guard statements.